blueprints/incredible-squaring/src/main.rs

```diff
@@ -8,139 +8,70 @@ use gadget_sdk::{
-
-        let xt = api::tx().services().register(
-            self.env.blueprint_id,
-            services::OperatorPreferences {
-                key: ecdsa::Public(ecdsa_pair.signer().public().0),
-                approval: services::ApprovalPrefrence::None,
-                // TODO: Set the price targets
-                price_targets: PriceTargets {
-                    cpu: 0,
-                    mem: 0,
-                    storage_hdd: 0,
-                    storage_ssd: 0,
-                    storage_nvme: 0,
-                },
-            },
-            Default::default(),
-        );
-
-        // send the tx to the tangle and exit.
-        let result = tx::tangle::send(&client, &signer, &xt).await?;
-        info!("Registered operator with hash: {:?}", result);
-        Ok(())
-    }
-
-    async fn benchmark(&self) -> std::result::Result<(), Self::Error> {
-        todo!()
-    }
 
-    async fn run(&mut self) -> Result<()> {
-        let client = self.env.client().await.map_err(|e| eyre!(e))?;
-        let signer = self.env.first_sr25519_signer().map_err(|e| eyre!(e))?;
+#[gadget_sdk::main(env)]
+async fn main() {
+    let client = env.client().await.map_err(|e| eyre!(e))?;
+    let signer = env.first_sr25519_signer().map_err(|e| eyre!(e))?;
 
-        info!("Starting the event watcher for {} ...", signer.account_id());
+    info!("Starting the event watcher for {} ...", signer.account_id());
 
-        let x_square = blueprint::XsquareEventHandler {
-            service_id: self.env.service_id.unwrap(),
-            client: client.clone(),
-            signer,
-        };
-
-        let finished_rx = x_square
-            .init_event_handler()
-            .await
-            .expect("Event Listener init already called");
-        let res = finished_rx.await;
-        gadget_sdk::error!("Event handler finished with {res:?}");
-
-        Ok(())
-    }
-}
-
-#[tokio::main]
-#[allow(clippy::needless_return)]
-async fn main() -> Result<()> {
-    gadget_sdk::logging::setup_log();
-    // Load the environment and create the gadget runner
-    let config = ContextConfig::from_args();
-    let env = gadget_sdk::config::load(config.clone()).expect("Failed to load environment");
-    let mut runner = Box::new(TangleGadgetRunner { env: env.clone() });
+    let x_square = blueprint::XsquareEventHandler {
+        service_id: env.service_id.unwrap(),
+        client: client.clone(),
+        signer,
+    };
 
     info!("~~~ Executing the incredible squaring blueprint ~~~");
-
-    check_for_test(&env, &config)?;
-
-    info!("Registering...");
-    // Register the operator if needed
-    if env.should_run_registration() {
-        runner.register().await?;
-    }
-
-    info!("Running...");
-    // Run the gadget / AVS
-    runner.run().await?;
+    MultiJobRunner::new(&env)
+        .with_job_context(env.clone())
+        .with_registration(registration)
+        .add_job(x_square)
```

Comment:
Naming nits:

`with_*` and `add_*` prefixes should be removed, builders typically just have `Builder::field()` methods. `with_*` is typically used for an associated function (like `MultiJobRunner::with_job_context(env).job(/* etc. */)`) ([example](https://rocket.rs/guide/v0.4/fairings/#attaching))